docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+data plane isolation is shared between Fybrik core, Fybrik users, module developers and probably IT teams.
+
+- Fybrik users should provide all relevant information to Fybrik and correctly configure their workloads. For example, 
+to set labels on the workload and inform Fybrik about the labels.
+
+- Depends on the isolation method, Fybrik should propagate all relevant information to the deployed modules, and/or 
+correctly configure Kubernetes Network Policies.
+- Module developers should allow to protect the module instances. For example, to allow TLS support
+
+In some complicated cases, IT teams interfere can be required, e.g. to configure external access points.
+
+## Requirements
+
+- Only predefined users/workloads should be able to access the data plane.
+- We do not assume that client workloads will be collocated on the same Kubernetes cluster with the data module, 
+furthermore they can run out of Kubernetes at all. 
+  - However, we do check the collocation of a Fybrik module and a relevant workload in the same cluster as a specific use case. 
+- We should allow workload client connections from different IP addresses. 
+
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). If a user workload or a Fybrik 
+previous chained module collocated with the next module in the same Kubernetes cluster, NPs can help to restrict the Fybrik 
+module ingress access. However, when a workload or a previous chained module runs out of the cluster, the NP based 
+isolation is going to be more complicated. The collocated deployment can be based on Kubernetes labels or namespaces, but 
+cross-clusters connections can be based on requester IP addresses, which is not always easy to discover. 
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
```

Comment:
This implementation is for isolating the workload from the module with which it interfaces.  Doesn't address isolating modules in the data plane I assume

Reply:
For chaining modules we don't need this, blueprint has all required information